### PR TITLE
Tpwdcee 787

### PR DESCRIPTION
**Pull request by rinnhofer**

### CODE REVIEW

File: woocommerce-wirecard-ee/classes/admin/class-wirecard-transaction-factory.php [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+	public function get_transaction_link( $base_url, $response ) {
+		$transaction_id = $response->getTransactionId();
+		$output         = 'For more transaction information click ';
+		$output        .= sprintf(
+			'<a target="_blank" href="' . $base_url . '/engine/rest/merchants/%s/payments/%s">',
+			$response->findElement( 'merchant-account-id' ),
+			$transaction_id
+		);
+		$output        .= 'here';
+		$output        .= '</a>';
```
Comment (jakubpolomsky) [inline woocommerce-wirecard-ee/classes/admin/class-wirecard-transaction-factory.php]: can we do it like this maybe?
`sprintf('% <a ...%...%>%</a>`) where the first and last % are for translated strings?

File: woocommerce-wirecard-ee/classes/handler/class-wirecard-response-handler.php [hunk 2]
```diff
@@ -57,10 +57,10 @@ class Wirecard_Response_Handler extends Wirecard_Handler {
 	public function handle_response( $request ) {
 		/** @var WC_Wirecard_Payment_Gateway $payment */
 		$payment = $this->get_payment_method( $request['payment-method'] );
-		$config  = $payment->create_payment_config();
+		$config  = $payment->create_payment_config();;
```
Comment (jakubpolomsky) [inline woocommerce-wirecard-ee/classes/handler/class-wirecard-response-handler.php]: obsolete `;`

File: woocommerce-wirecard-ee/classes/handler/class-wirecard-transaction-handler.php [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+		if ( $response instanceof SuccessResponse ) {
+			$order = wc_get_order( $transaction_data->order_id );
+			$order->set_transaction_id( $response->getTransactionId() );
+			$redirect_url = '/admin.php?page=wirecardpayment&id=' . $response->getTransactionId();
+			wp_redirect( admin_url( $redirect_url ), 301 );
+			die();
+		}
+		if ( $response instanceof FailureResponse ) {
+			echo 'failed to cancel';
```
Comment (jakubpolomsky) [inline woocommerce-wirecard-ee/classes/handler/class-wirecard-transaction-handler.php]: translation please ^^

File: woocommerce-wirecard-ee/classes/handler/class-wirecard-transaction-handler.php [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+		if ( $response instanceof SuccessResponse ) {
+			$order = wc_get_order( $transaction_data->order_id );
+			$order->set_transaction_id( $response->getTransactionId() );
+			$redirect_url = '/admin.php?page=wirecardpayment&id=' . $response->getTransactionId();
+			wp_redirect( admin_url( $redirect_url ), 301 );
+			die();
+		}
+		if ( $response instanceof FailureResponse ) {
+			echo 'failed to capture';
```
Comment (jakubpolomsky) [inline woocommerce-wirecard-ee/classes/handler/class-wirecard-transaction-handler.php]: translation please ^^

Comment (rinnhofer) [reply woocommerce-wirecard-ee/classes/handler/class-wirecard-transaction-handler.php]: done

File: woocommerce-wirecard-ee/classes/handler/class-wirecard-transaction-handler.php (hunk 3 above)
Comment (rinnhofer) [reply woocommerce-wirecard-ee/classes/handler/class-wirecard-transaction-handler.php]: done

File: woocommerce-wirecard-ee/classes/handler/class-wirecard-response-handler.php (hunk 2 above)
Comment (rinnhofer) [reply woocommerce-wirecard-ee/classes/handler/class-wirecard-response-handler.php]: done

File: woocommerce-wirecard-ee/classes/admin/class-wirecard-transaction-factory.php (hunk 1 above)
Comment (rinnhofer) [reply woocommerce-wirecard-ee/classes/admin/class-wirecard-transaction-factory.php]: done

**Pull request merged**